mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergSerDe.java

```diff
@@ -66,9 +67,7 @@ public void initialize(@Nullable Configuration configuration, Properties serDePr
     assertNotVectorizedTez(configuration);
 
     Schema tableSchema;
-    if (configuration.get(InputFormatConfig.TABLE_SCHEMA) != null) {
-      tableSchema = SchemaParser.fromJson(configuration.get(InputFormatConfig.TABLE_SCHEMA));
-    } else if (serDeProperties.get(InputFormatConfig.TABLE_SCHEMA) != null) {
+    if (serDeProperties.get(InputFormatConfig.TABLE_SCHEMA) != null) {
```

Comment:
Is this needed?
The original intent of the change was that we have the table schema at hand on the mappers/reducers. If we remove this then every mapper/reducer has to read the table once to get the schema.

Reply:
As show below:
* Mapper handles input split belongs to one table: 
https://github.com/apache/hive/blob/113f6af7528f016bf821f7a746bad496cc93f834/ql/src/java/org/apache/hadoop/hive/ql/io/HiveInputFormat.java#L406
* Function **copyTableJobPropertiesToConf** copies 'iceberg.mr.table.schema' property to jobConf:
https://github.com/apache/hive/blob/113f6af7528f016bf821f7a746bad496cc93f834/ql/src/java/org/apache/hadoop/hive/ql/exec/Utilities.java#L2427-L2443
 
if we join two tables, only one iceberg schema exists in jobConf which leads wrong inspector:
* empty inspector(non-overlap in two table selected columns, e.g. 
`SELECT o.order_id, o.customer_id, o.total, p.name FROM default.orders o JOIN default.products p ON o.product_id = p.id ORDER BY o.order_id`
selected columns in table default.orders: [order_id, total, customer_id, product_id]
selected columns in table default.products: [id, name])
* incomplete inspector(overlap in two table selected columns, e.g.
`SELECT c.first_name, o.order_id FROM default.orders o JOIN default.customers c ON o.customer_id = c.customer_id ORDER BY o.order_id DESC`
selected columns in table default.customers: [first_name, customer_id]
selected columns in table default.orders: [order_id, customer_id]

Reply:
Thanks for the explanation! This is something @marton-bod already tried his hands on. We might have to use the table identifier to prefix/postfix the schema....

Reply:
Hi @marton-bod , is there any thread work on the proposal @pvary mentioned above? I can not find any related issue on github. 
Thanks.

Reply:
Hi @qphien . Indeed, we have had a similar problem before in https://github.com/apache/iceberg/issues/1708. 
My initial solution to fix it was to add the schema into the config object with a prefix based on the table identifier. For example, when joining together default.orders and default.customers, you'd have two properties in the config: `default.orders.iceberg.mr.table.schema` and `default.customers.iceberg.mr.table.schema`. This should allow you to add the schema for multiple tables without collisions/overwrites.

Eventually we ended up not adding in this fix, but instead just reverted the commit that caused the problem for us. However, this prefix solution can be one way to make things work.

Reply:
I think getting table schema from `serDeProperties` has the same effect as prefix solution. `iceberg.mr.table.schema` in `serDeProperties` is overlayed from corresponding tableDesc, so we can get correct table schema when `HiveIcebergSerde` is initialized.